src/neo/SmartContract/Native/ManagementContract.cs

```diff
@@ -210,10 +214,21 @@ private void Destroy(ApplicationEngine engine)
             StorageKey ckey = CreateStorageKey(Prefix_Contract).Add(hash);
             ContractState contract = engine.Snapshot.Storages.TryGet(ckey)?.GetInteroperable<ContractState>();
             if (contract is null) return;
+            GetContractSet(engine.Snapshot).Remove(hash);
             engine.Snapshot.Storages.Delete(ckey);
             foreach (var (key, _) in engine.Snapshot.Storages.Find(BitConverter.GetBytes(contract.Id)))
                 engine.Snapshot.Storages.Delete(key);
             engine.SendNotification(Hash, "Destory", new VM.Types.Array { hash.ToArray() });
         }
+
+        public Dictionary<UInt160, ContractState> GetContractSet(StoreView snapshot)
+        {
+            if (snapshot.ContractSet == null)
+            {
+                byte[] listContractsPrefix = CreateStorageKey(Prefix_Contract).ToArray();
+                snapshot.ContractSet = snapshot.Storages.Find(listContractsPrefix).Select(kvp => kvp.Value.GetInteroperable<ContractState>()).ToDictionary(p => p.Hash);
+            }
+            return snapshot.ContractSet;
+        }
```

Comment:
I think it is very slow to read all the contracts in each block.

Reply:
Another thought is to create a Prefix_ContractSet to store all contract hashes in database, and updated synchronized with Prefix_Contract. Is that one better?

Reply:
Can we use a bloomfilter?

Reply:
We have already adopted bloomfilter in leveldb, but it's not suitable for this..Currently in func `PostTransfer` we need to check  whether the receiver is a contract or not. The receiver can be any address.

Reply:
Aftering modification we don't need to search every block